pkg/asset/installconfig/gcp/gcp.go

```diff
@@ -82,13 +81,36 @@ func selectProject(ctx context.Context) (string, error) {
 	return selectedProject, err
 }
 
-func selectRegion(project string) (string, error) {
-	longRegions := make([]string, 0, len(validation.Regions))
-	shortRegions := make([]string, 0, len(validation.Regions))
-	for id, location := range validation.Regions {
-		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
-		shortRegions = append(shortRegions, id)
+func selectRegion(ctx context.Context, project string) (string, error) {
+
+	ssn, err := GetSession(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get session")
+	}
+
+	client := &Client{
+		ssn: ssn,
+	}
+
+	validRegions, err := client.GetRegions(ctx, project)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get regions")
+	}
+
+	defaultRegion := "us-central1"
+	defaultRegionName := ""
+	longRegions := make([]string, 0, len(validRegions))
+	shortRegions := make([]string, 0, len(validRegions))
+	for key, value := range validRegions {
+		shortRegions = append(shortRegions, key)
+		regionDesc := fmt.Sprintf("%s (%s)", key, value)
+		longRegions = append(longRegions, regionDesc)
+
+		if key == defaultRegion {
+			defaultRegionName = regionDesc
+		}
```

Comment:
It might be worth using a structure here instead of using three different variables.